Tests/Opc.Ua.Core.Tests/Types/ContentFillter/ContentFilterTests.cs

```diff
@@ -0,0 +1,422 @@
+ * ======================================================================*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using Opc.Ua.Test;
+
+namespace Opc.Ua.Core.Tests.Types.ContentFillter
+{
+
+    internal class TestFilterTarget : IFilterTarget
+    {
+        public object GetAttributeValue(FilterContext context, NodeId typeDefinitionId, IList<QualifiedName> relativePath, uint attributeId, Ua.NumericRange indexRange)
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool IsTypeOf(FilterContext context, NodeId typeDefinitionId)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    /// <summary>
+    /// Tests for the BuiltIn Types.
+    /// </summary>
+    [TestFixture, Category("ContentFillter")]
+    [SetCulture("en-us"), SetUICulture("en-us")]
+    [Parallelizable]
+    public class ContentFilterTests
+    {
+        #region Shared Properties
+        public EventFilter Filter { get;}
+        public FilterContext FilterContext { get;}
+        #endregion
+
+        #region Constructor
+        public ContentFilterTests()
+        {
+            // construct the context to use for the event filter.
+            NamespaceTable namespaceTable = new NamespaceTable();
+            TypeTable typeTable = new TypeTable(namespaceTable);
+            FilterContext = new FilterContext(namespaceTable, typeTable);
+
+            // event filter must be specified.
+            Filter = new EventFilter();
+            Filter.WhereClause = new ContentFilter();
+        }
+        #endregion
+
+        #region Test Setup
+        [OneTimeSetUp]
+        protected void OneTimeSetUp()
+        {
+        }
+
+        [OneTimeTearDown]
+        protected void OneTimeTearDown()
+        {
+        }
+
+        [SetUp]
+        protected void SetUp()
+        {
+        }
+
+        [TearDown]
+        protected void TearDown()
+        {
+        }
+        #endregion
+
+        #region Test Methods
+
+        [Test]
+        [Category("ContentFillter")]
+        public void And()
+        {
+            LiteralOperand loperand1 = new LiteralOperand();
+            LiteralOperand loperand2 = new LiteralOperand();
+
+            loperand1.Value = new Variant(true);
+            loperand2.Value = new Variant(false);
+
+            ContentFilterElement filterElement = new ContentFilterElement();
+            filterElement.FilterOperator = FilterOperator.And;
+            filterElement.SetOperands(new List<LiteralOperand>() { loperand1, loperand2 });
+            Filter.WhereClause.Elements = new[] { filterElement };
+
+            TestFilterTarget filterTarget = new TestFilterTarget();
+            // apply filter.
+            object result = Filter.WhereClause.Evaluate(FilterContext, filterTarget);
+            Assert.AreEqual(false, result);
+        }
+
+        [Test]
+        [Category("ContentFillter")]
```

Comment:
I wonder if some of the tests coud be simplified using the 'TestParameter' attribute. Also I miss the positive/negative test cases and a test for multiple filter elements, when the filter operators are used one by one.

Reply:
Tests now use the TestCase attribute